### DISCUSSION TITLE
Vector field position after reading and writing a mesh

### DISCUSSION THREAD

**Discussion by mabe2416** [simnibs/simnibs#329] [9. Results and analysis] [answered]
Hello

When I'm reading and then writing a mesh (without doing anything in between), the position from the vectors J are not shown in the centre of the elements anymore but in the nodes. 

Here is my python code:
mesh: sm.Msh = sm.read_msh(simMesh+'.msh', m=None)
mesh.write(simMesh+'2.msh')

And this shows the triangles of a surface with J in gmsh.
Original mesh:
![J_original](https://github.com/simnibs/simnibs/assets/159785086/415819e6-453d-487a-84b5-17bc3fa01ecc)
After read and write: 
![J_after_write](https://github.com/simnibs/simnibs/assets/159785086/463adb74-a827-4e35-a3fa-d4463ff54dab)

I have simnibs 4.1.0a installed.
Does anyone know, whats the cause and how to fix this?

**Comment by mabe2416** [comment]
It must be related to msh.write(). 
If I use mesh.view().show() of the loaded mesh file, the position of the vectors is still in the element centre.

**Comment by TorgeW** [comment, marked as answer]
So to me this sounds like that you have different settings in gmsh. 
Have you tried writing the mesh out and reading it back in with simnibs, looking at the show() again. It should look the same after laoding it back in.

There are some settings in gmsh (the opt file) that change where the vectors are displayed:
Options -> Aspect 

The other thing you can check is, after writing it out and back in, is the field still an element data filed or a node data field. 

But all in all I am guessing its just something with the displaying in gmsh.

**Reply by mabe2416** [reply]
Thanks! I'll test it out.
